consensus/polybft/stake_manager.go

```diff
@@ -115,12 +121,23 @@ func (s *stakeManager) PostBlock(req *PostBlockRequest) error {
 		return err
 	}
 
-	s.logger.Debug("Stake manager on post block", "block", req.FullBlock.Block.Number(),
-		"last saved", fullValidatorSet.BlockNumber, "last updated", fullValidatorSet.UpdatedAtBlockNumber)
+	s.logger.Debug("Stake manager on post block",
+		"block", req.FullBlock.Block.Number(),
+		"last saved", fullValidatorSet.BlockNumber,
+		"last updated", fullValidatorSet.UpdatedAtBlockNumber)
 
-	err = s.updateWithReceipts(&fullValidatorSet, req.FullBlock)
+	// get transfer currentBlockEvents from current block
+	transferEvents, err := s.eventsGetter.getFromBlocks(fullValidatorSet.BlockNumber, req.FullBlock)
 	if err != nil {
-		return err
+		return fmt.Errorf("could not get transfer events from current block. Error: %w", err)
+	}
+
+	// we should save new state even if number of events is zero
+	// because otherwise next time we will process more blocks
+	if len(transferEvents) > 0 {
```

Comment:
Couldn't you add early return inside `updateWithReceipt` function in case `transferEvents` are empty? Then you can get rid off this check on two places.

Reply:
I wanted to save state in any case (comment above explains it)

Reply:
fixed https://github.com/0xPolygon/polygon-edge/pull/1928/commits/7459127e696d97266ca729acc9ffa7de07762236